src/components/application_manager/test/commands/command_request_impl_test.cc

```diff
@@ -0,0 +1,485 @@
+  const std::string kDisallowedParam1 = "disallowed_param1";
+  const std::string kDisallowedParam2 = "disallowed_param2";
+  const std::string kAlParam = "allowed_param";
+  const std::string kUnParam = "undefined_params";
+  const std::string kMisParam =
+      ::application_manager::MessageHelper::vehicle_data().begin()->first;
+
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::msg_params][kDisallowedParam1] = 0u;
+  (*msg)[strings::msg_params][kDisallowedParam2] = 0u;
+  (*msg)[strings::msg_params][kAlParam] = 0u;
+  (*msg)[strings::msg_params][kUnParam] = 0u;
+  (*msg)[strings::msg_params][kMisParam] = 0u;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  CommandParametersPermissions& permission = command->parameters_permissions();
+  permission.disallowed_params.push_back(kDisallowedParam1);
+  permission.disallowed_params.push_back(kDisallowedParam2);
+  permission.allowed_params.push_back(kAlParam);
+  permission.undefined_params.push_back(kUnParam);
+
+  command->RemoveDisallowedParameters();
+
+  EXPECT_FALSE((*msg)[strings::msg_params].keyExists(kDisallowedParam1));
+  EXPECT_FALSE((*msg)[strings::msg_params].keyExists(kDisallowedParam2));
+  EXPECT_FALSE((*msg)[strings::msg_params].keyExists(kUnParam));
+  EXPECT_FALSE((*msg)[strings::msg_params].keyExists(kMisParam));
+  EXPECT_TRUE((*msg)[strings::msg_params].keyExists(kAlParam));
+  EXPECT_TRUE(command->HasDisallowedParams());
+}
+
+TEST_F(CommandRequestImplTest,
+       CheckAllowedParameters_RegisterAppInterface_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::function_id] =
+      mobile_apis::FunctionID::RegisterAppInterfaceID;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  EXPECT_CALL(app_mngr_, applications()).Times(0);
+  EXPECT_TRUE(command->CheckPermissions());
+}
+
+TEST_F(CommandRequestImplTest,
+       CheckAllowedParameters_NoAppWithSameConnectionKey_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey_;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+  ApplicationSet app_set;
+  app_set.insert(app);
+
+  DataAccessor<ApplicationSet> accessor(app_set, app_set_lock_);
+
+  EXPECT_CALL(app_mngr_, applications()).WillOnce(Return(accessor));
+  EXPECT_CALL(*app, app_id()).WillOnce(Return(6u));
+  EXPECT_TRUE(command->CheckPermissions());
+}
+
+TEST_F(CommandRequestImplTest, CheckAllowedParameters_NoMsgParamsMap_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey_;
+  (*msg)[strings::msg_params] = 0u;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+  ApplicationSet app_set;
+  app_set.insert(app);
+
+  DataAccessor<ApplicationSet> accessor(app_set, app_set_lock_);
```

Comment:
@LevchenkoS May be move 3 code line above into separate function which returns `DataAccessor<ApplicationSet>`. You are using the same 3-lines piece of code in lots of tests cases.